nativelib/src/main/resources/stat.c

```diff
@@ -0,0 +1,168 @@
+#include <sys/stat.h>
+#include <gc.h>
+#include "types.h"
+
+// We don't use the "standard" types such as `dev_t` for instance
+// because these have different sizes on eg. Linux and OSX. We use the
+// smallest type that can hold all the possible values for the different systems.
```

Comment:
This is acceptable solution, as long as we consistently copy original C's `stat` to our stat in all of our bindings.